File: python-bindings/tests/crash_pattern_conformance.py
"""Input-only public Python crash-token observations."""

from collections.abc import Mapping
from typing import Any


def observe_crash_pattern(fixture: Mapping[str, Any]) -> dict[str, Any]:
    """Read the real shared native classifier result without host token inference."""
    from classic_scanlog import detect_crash_pattern

    if fixture["operation"] == "vr":
        from classic_scanlog import LogParser

        return {"vr": LogParser().detect_vr_log(fixture["content"])}
    if fixture["operation"] != "classify":
        raise ValueError("Python does not expose the legacy CXX error-text alias")
    return {"token": detect_crash_pattern(fixture["content"])}

File: node-bindings/classic-node/__test__/crash_pattern_conformance.ts
import { detectCrashPattern, detectVrLog } from "../index.js";

/** Observe the actual classifier token; no adapter-local pattern table is allowed. */
export function observeCrashPattern(fixture:Record<string,any>):Record<string,any>{
  if(fixture.operation==="vr")return {vr:detectVrLog(fixture.content)};
  if(fixture.operation!=="classify")throw new Error("Node has no legacy CXX error-text alias");
  return {token:detectCrashPattern(fixture.content) ?? null};
}
